### DISCUSSION TITLE
How do I declare a contract address as an address/contract, rather than a string?

### DISCUSSION THREAD

**Discussion by salieflewis** [wevm/wagmi#1683] [Q&A]
I have a simple interface `ManagerReturnTypes`

```
export interface ManagerReturnTypes {
  tokenAddress?: string
  daoAddresses: {
    metadataAddress: string
    auctionAddress: string
    treasuryAddress: string
    governorAddress: string
  }
}
```

Elsewhere in my code, I need to call `queryFilter` on one of those addresses, but before I do so, I need to safely type it as a contract.

```
const governorContract = getContract({
    addressOrName: governorAddress,
    contractInterface: governorAbi,
  })
```
How can I just provide that type definition off the bat when I'm defining my interface?

**Comment by tmm** [comment]
Not sure what you are asking. ABIType doesn't have any notion of `queryFilter`, `getContract`, etc. Are you trying to use the wagmi `getContract` types inside your `ManagerReturnTypes`?

**Reply by salieflewis** [reply]
Yes. Instead of defining those addresses as strings initially. I just want them to be Contract types. Hope that makes more sense.

**Reply by tmm** [reply]
Got it. This is more of a wagmi question than ABIType question so transferring this discussion over there. Do you already have ABIs for these contracts? If so, you can do something like this:

```ts
import type { GetContractResult } from '@wagmi/core'

export interface ManagerReturnTypes {
  tokenAddress?: GetContractResult<typeof tokenAbi>
  daoAddresses: {
    metadataAddress: GetContractResult<typeof metadataAbi>
    auctionAddress: GetContractResult<typeof auctionAbi>
    treasuryAddress: GetContractResult<typeof treasuryAbi>
    governorAddress: GetContractResult<typeof governorAbi>
  }
}
```

**Reply by salieflewis** [reply]
Amazing! That is super helpful, thanks. So if you don't mind me asking, in what scenarios are you reaching for abitype?

**Reply by tmm** [reply]
ABIType is for lower-level type inference for ABIs and EIP-712 Typed Data. It doesn't have a concept of a "Contract" (e.g. `GetContractResult` has a lot of extra types).
